admin/src/app/home/orders/stock-orders/receive/receive.component.ts

```diff
@@ -53,216 +62,294 @@ export class ReceiveComponent implements OnInit, OnDestroy {
   public discrepancyOrderItem: any;
   private subscriptions: Subscription[] = [];
   public sortAscending = true;
-  public sortColumn = 'productModelSku';
+  public sortColumn = "productModelSku";
   showAddProductModal: boolean;
   public bsModalRef: BsModalRef;
   public isSmallDevice = this.sharedDataService.getIsSmallDevice();
   public isDiscrepancyLoaded = false;
   public editingDamagedForItemId;
   public damagedQuantity = 0;
 
-  @ViewChild('discrepancies') discrepanciesTab;
-  sendDiscrepancyReport: any = 'true';
+  @ViewChild("discrepancies") discrepanciesTab;
+  sendDiscrepancyReport: any = "true";
   selectedCategoryLabelFilter: string;
 
-  constructor(private orgModelApi: OrgModelApi,
-              private _route: ActivatedRoute,
-              private _router: Router,
-              private toastr: ToastrService,
-              private _userProfileService: UserProfileService,
-              private _eventSourceService: EventSourceService,
-              private auth: LoopBackAuth,
-              private modalService: BsModalService,
-              private sharedDataService: SharedDataService) {
-  }
+  constructor(
+    private orgModelApi: OrgModelApi,
+    private _route: ActivatedRoute,
+    private _router: Router,
+    private toastr: ToastrService,
+    private _userProfileService: UserProfileService,
+    private _eventSourceService: EventSourceService,
+    private auth: LoopBackAuth,
+    private modalService: BsModalService,
+    private sharedDataService: SharedDataService,
+    private barcodeReceiveService: BarcodeReceiveService
+  ) {}
 
   ngOnInit() {
     this.userProfile = this._userProfileService.getProfileData();
-    this._route.data.subscribe((data: any) => {
+    this._route.data.subscribe(
+      async (data: any) => {
         this.order = data.stockOrderDetails[0];
+
+        // 1. Delete all line items
+        await productDB.products.clear();
```

Comment:
Can we use dexie database class/service to have these  functions 

and call it through the database class/service